### PR TITLE
feat: support extra javadoc tags

### PR DESCRIPTION
**Pull request by alandefreitas**
This PR includes support for tags that are used in Boost.URL but are currently ignored by MrDocs.

### CODE REVIEW

File: include/mrdocs/Metadata/Javadoc.hpp [hunk 1]
```diff
@@ -172,7 +242,7 @@ struct Text : Node
     bool equals(const Node& other) const noexcept override
     {
         return kind == other.kind &&
-            *this == static_cast<const Text&>(other);
+            *this == dynamic_cast<const Text&>(other);
```
Comment (sdkrystian) [inline include/mrdocs/Metadata/Javadoc.hpp:245]: Why did you change these to `dynamic_cast`? `kind == other.kind` implies that `other` is derived from `Text`.

File: src/lib/AST/ParseJavadoc.cpp [hunk 2]
```diff
@@ -296,65 +474,126 @@ visitTextComment(
...
+
+    // Find Comment::HTMLEndTagCommentKind
+    auto const tagEndIt =
+        requiresEndTag ? std::ranges::find_if(it_ + 1, end_, [](Comment const* c)
+        {
+            return c->getCommentKind() == Comment::HTMLEndTagCommentKind;
+        }) : it_;
+    if (tagEndIt == end_)
+    {
+        return Unexpected(Error(fmt::format("warning: HTML <{}> tag not followed by end tag", res.tag)));
```
Comment (sdkrystian) [inline src/lib/AST/ParseJavadoc.cpp:501]: Could just use `formatError` here

File: src/lib/Gen/adoc/AdocGenerator.cpp [hunk 3]
```diff
@@ -75,7 +75,7 @@ build(
 
     auto errors = ex->wait();
     if(! errors.empty())
-        return Error(errors);
+        return {errors};
```
Comment (sdkrystian) [inline src/lib/Gen/adoc/AdocGenerator.cpp:78]: Please forgo stylistic changes unless you're making functional changes to that code

File: src/lib/AST/ParseJavadoc.cpp [hunk 4]
```diff
@@ -4,6 +4,7 @@
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
+// Copyright (c) 2024 Alan de Freitas (alandefreitas@gmail.com)
```
Comment (sdkrystian) [inline src/lib/AST/ParseJavadoc.cpp:7]: Minor nit: copyright holders should be in chronological order

File: include/mrdocs/Metadata/Javadoc.hpp (hunk 1 above)
Comment (alandefreitas) [reply include/mrdocs/Metadata/Javadoc.hpp:245]: It's a warning I get.

> Do not use static_cast to downcast from a base to a derived class; use dynamic_cast instead

https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/pro-type-static-cast-downcast.html

File: src/lib/Gen/adoc/AdocGenerator.cpp (hunk 3 above)
Comment (alandefreitas) [reply src/lib/Gen/adoc/AdocGenerator.cpp:78]: That was also a pedantic warning that Vinnie also always told me apply, but OK.

https://clang.llvm.org/extra/clang-tidy/checks/modernize/return-braced-init-list.html
